### ISSUE TITLE
Fixed Chronicle-Queue on jdk11  with  utf8

### ISSUE THREAD

**Issue by whitelilis** [OpenHFT/Chronicle-Queue#553]
I create a demo to reproduce it  : https://github.com/whitelilis/Chronicle-Queue-jdk11-utf8-bug

on jdk11 
1, run the Reader
2, run the Writer
the Reader will got value 0, but not expected 15

on jdk1.8 , it work well

**Comment by whitelilis** [comment]
Is there some work on this issue?

**Comment by RobAustin** [comment]
Nope - Not at the moment as we are working on higher priority tasks. If you would like this issue to be given a higher priority, please purchase a support agreement from sales@chronicle.software

**Comment by RobAustin** [comment]
Alternatively, if you would like to fix it, if you issue a pull request we can merge in your change.

**Comment by MartyIX** [comment]
I have taken a quick look and I have found that this line https://github.com/OpenHFT/Chronicle-Wire/blob/master/src/main/java/net/openhft/chronicle/wire/BinaryWire.java#L1289 is not correct. It does not take into account JDK 9+ where it is necessary to use extractBytes method (https://github.com/OpenHFT/Chronicle-Core/blob/master/src/main/java/net/openhft/chronicle/core/util/StringUtils.java#L201).

I'll try to dig deeper :-)

**Comment by MartyIX** [comment]
Well, so the problem is in the if statement: https://github.com/OpenHFT/Chronicle-Wire/blob/master/src/main/java/net/openhft/chronicle/wire/BinaryWire.java#L1288.

1) When I remove the "if" statement completely, it starts working. Of course, it's not desirable thing to do in general.

note: Changing `blob.mm = "你好";` to a longer string `blob.mm = "你好你好你好你好你好";` makes it work too as the conditional statement is not true. 

2) When I try to hot-fix the issue using

```
} else if (len < 0x20) {
    if (Jvm.isJava9Plus()) {
        byte coder = StringUtils.getStringCoder(s);
        byte[] strBytes = extractBytes(s);
        long utfLength = AppendableUtil.findUtf8Length(strBytes, coder);
        bytes.writeUnsignedByte((int) (STRING_0 + utfLength)).appendUtf8(strBytes, 0, s.length(), coder);
    } else {
        bytes.writeUnsignedByte((int) (STRING_0 + len)).appendUtf8(StringUtils.extractChars(s), 0, s.length());
    }

```

the bug is still present. It seems there is a difference in parsing STRING_0 and STRING_ANY in terms of unicode support or something along these lines.

**Comment by RobAustin** [comment]
@MartyIX thanks for investigating this, we really appreciate it, would you be able to submit a pull request with a fix.

**Comment by MartyIX** [comment]
@RobAustin Yes, I'll submit a pull request. I'll just need a bit more time to understand a few things better.

**Comment by hft-team-city** [comment]
Released in [Chronicle-Queue-5.17.4](https://github.com/OpenHFT/Chronicle-Queue/releases/chronicle-queue-5.17.4), [BOM-2.17.126](https://github.com/OpenHFT/OpenHFT/releases/chronicle-bom-2.17.126)

**Comment by hft-team-city** [comment]
Released in [Chronicle-Queue-5.17.4](https://github.com/OpenHFT/Chronicle-Queue/releases/chronicle-queue-5.17.4), [BOM-2.17.196](https://github.com/OpenHFT/OpenHFT/releases/chronicle-bom-2.17.196)

**Comment by hft-team-city** [comment]
Released in [Chronicle-Queue-5.17.4](https://github.com/OpenHFT/Chronicle-Queue/releases/chronicle-queue-5.17.4), [BOM-2.17.225](https://github.com/OpenHFT/OpenHFT/releases/chronicle-bom-2.17.225)

**Comment by hft-team-city** [comment]
Released in [Chronicle-Queue-5.17.4](https://github.com/OpenHFT/Chronicle-Queue/releases/chronicle-queue-5.17.4), [BOM-2.17.241](https://github.com/OpenHFT/OpenHFT/releases/chronicle-bom-2.17.241)
